crates/languages/src/common/python_common.rs

```
@@ -0,0 +1,236 @@
+                // default_parameter structure: identifier = value
+                let name = child
+                    .named_child(0)
+                    .and_then(|n| {
+                        if n.kind() == "identifier" {
+                            node_to_text(n, source).ok()
+                        } else {
+                            None
+                        }
+                    })
+                    .unwrap_or_default();
+                parameters.push((name, None));
+            }
+            "typed_default_parameter" => {
+                // typed_default_parameter structure: identifier : type = value
+                let name = child
+                    .named_child(0)
+                    .and_then(|n| {
+                        if n.kind() == "identifier" {
+                            node_to_text(n, source).ok()
+                        } else {
+                            None
+                        }
+                    })
+                    .unwrap_or_default();
+                let type_hint = child
+                    .child_by_field_name("type")
+                    .and_then(|n| node_to_text(n, source).ok());
+                parameters.push((name, type_hint));
+            }
+            "list_splat_pattern" => {
+                let name = child
+                    .named_child(0)
+                    .and_then(|n| node_to_text(n, source).ok())
+                    .map(|n| format!("*{n}"))
+                    .unwrap_or_else(|| "*args".to_string());
+                parameters.push((name, None));
+            }
+            "dictionary_splat_pattern" => {
+                let name = child
+                    .named_child(0)
+                    .and_then(|n| node_to_text(n, source).ok())
+                    .map(|n| format!("**{n}"))
+                    .unwrap_or_else(|| "**kwargs".to_string());
+                parameters.push((name, None));
+            }
+            _ => {}
+        }
+    }
```

The catch-all `_ => {}` silently ignores Python 3.8+ positional-only (`def f(a, /, b)`) and keyword-only (`def f(*, a)`) parameters. Consider adding explicit handling or at least logging when unknown parameter types are encountered.